packages/widgets/src/lib/components/InferenceWidget/shared/helpers.ts

```diff
@@ -173,11 +173,19 @@ export async function getModelLoadInfo(
 	const response = await fetch(`${url}/status/${repoId}`, {
 		credentials: includeCredentials ? "include" : "same-origin",
 	});
-	const output = await response.json();
+	const output: {
+		state: LoadState;
+		compute_type: ComputeType | Record<ComputeType, { [key in ComputeType]?: string } & { count: number }>;
+		loaded: boolean;
+		error: Error;
+	} = await response.json();
 	if (response.ok && typeof output === "object" && output.loaded !== undefined) {
 		// eslint-disable-next-line @typescript-eslint/naming-convention
-		const { state, compute_type } = output;
-		return { compute_type, state };
+		const compute_type =
+			typeof output.compute_type === "string"
+				? output.compute_type
+				: (Object.keys(output.compute_type)[0] as ComputeType);
+		return { compute_type, state: output.state };
```

Comment:
lol the shape is a bit unpractical to work with

nit (feel free to ignore): we could favor `gpu` over `cpu` if present


```suggestion
		const compute_type =
			typeof output.compute_type === "string"
				? output.compute_type
				: output.compute_type["gpu"] ? ("gpu" as const) : ("cpu" as const);
		return { compute_type, state: output.state };
```